meteor/lib/collections/Rundowns.ts

```diff
@@ -163,17 +163,64 @@ export class Rundown implements DBRundown {
 			}, options)
 		).fetch()
 	}
-	getParts (selector?: MongoSelector<Part>, options?: FindOptions) {
+	getParts (selector?: MongoSelector<Part>, options?: FindOptions, segmentsInOrder?: Segment[]): Part[] {
 		selector = selector || {}
 		options = options || {}
-		return Parts.find(
+
+		let parts = Parts.find(
 			_.extend({
 				rundownId: this._id
 			}, selector),
 			_.extend({
 				sort: { _rank: 1 }
 			}, options)
 		).fetch()
+		if (!options.sort) {
+			parts = Rundown._sortParts(parts, segmentsInOrder || this.getSegments())
+		}
+		return parts
+	}
+	/**
+	 * Return ordered lists of all Segments and Parts in the rundown
+	 */
+	async getSegmentsAndParts (): Promise<{ segments: Segment[], parts: Part[] }> {
+
+		const pSegments = asyncCollectionFindFetch(Segments, {
+			rundownId: this._id
+		}, { sort: { _rank: 1 } })
+
+		const pParts = asyncCollectionFindFetch(Parts, {
+			rundownId: this._id
+		}, { sort: { _rank: 1 } })
+
+		const segments = await pSegments
+		return {
+			segments: segments,
+			parts: Rundown._sortParts(await pParts, segments)
+		}
+	}
+	/** Synchronous version of getSegmentsAndParts, to be used client-side */
+	getSegmentsAndPartsSync (): { segments: Segment[], parts: Part[] } {
```

Comment:
I'm starting to have second thoughts about this: won't such generic queries cause Trackers to refresh whenever there is any change in any of the parts, not just when Parts in the segment change? Also, wouldn't it make sense to filter parts at this stage instead of doing two passes though the Parts arrays first with the rundown filter and then with the segment filter?